File: app/src/main/java/com/jackyli/androidarchitecture/api/ApiService.java
package com.jackyli.androidarchitecture.api;

import com.jackyli.androidarchitecture.model.Result;
import com.jackyli.androidarchitecture.model.UserInfo;

import io.reactivex.Observable;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiService {

    //登录
    @GET("login")
    Call<Result> loginNetwork(@Query("userName") String name, @Query("passWord") String password);
}

File: app/src/main/java/com/jackyli/androidarchitecture/api/ApiService.kt
package com.jackyli.androidarchitecture.api

import com.jackyli.androidarchitecture.model.Result
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiService {

    @GET("login")
    fun loginNetWork(@Query("userName") name: String, @Query("passWord") pass: String): Call<Result>
}